includes/rest-api.php

```diff
@@ -338,6 +405,125 @@ function check_post_types_permissions() {
 	return $response;
 }
 
+/**
+ * Get a list of content to show on the Pull screen
+ *
+ * @param \WP_Rest_Request $request API request arguments
+ * @return \WP_REST_Response|\WP_Error
+ */
+function get_pull_content( $request ) {
+	$args = [
+		'posts_per_page' => isset( $request['per_page'] ) ? $request['per_page'] : 10,
+		'paged'          => isset( $request['page'] ) ? $request['page'] : 1,
+		'post_type'      => isset( $request['post_type'] ) ? $request['post_type'] : 'post',
+		'post_status'    => isset( $request['post_status'] ) ? $request['post_status'] : array( 'any' ),
+	];
+
+	if ( ! empty( $request['search'] ) ) {
+		$args['s'] = urldecode( $request['search'] );
+	}
+
+	if ( ! empty( $request['exclude'] ) ) {
+		$args['post__not_in'] = $request['exclude'];
+	}
+
+	// Filter documented in includes/classes/ExternalConnections/WordPressExternalConnection.php
+	$query = new \WP_Query( apply_filters( 'dt_remote_get_query_args', $args, $request ) );
```

Comment:
I've gone ahead and removed this filter as it isn't needed. We run the same filter before we end up making our remote POST request, so this would just result in filtering the same values twice.

Reply:
And after thinking about it a bit more, not sure it makes sense to run the first filter, as that's meant to filter the remote_get query args. I've moved that code to run after our remote post call and added a new filter in the remote post method around those arguments. So to answer your second question, if you only want to filter the arguments for a Pull call, you'd use this new filter instead